### PR TITLE
Feat/provision mfl

### PR DESCRIPTION
**Pull request by MohamedAliBouhaouala**
## Type of PR (check all applicable)

- [ ] 💼 New Package
- [x] ♻️ Refactor
- [x] ✨ New Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🧪 Tests
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

This PR provisions MFL location and organization FHIR resources into Hapi FHIR

## Tickets & Related Documents

Resolves #

## Added code snippets?
- [ ] 👍 Yes
- [x] 🙅 No, because they aren't needed

## Added tests?

- [ ] 👍 Yes
- [x] 🙅 No, because they aren't needed
- [ ] 🙋 No, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📓 general documentation
- [x] 🙅 no documentation needed

## Are there any post-deployment tasks we need to perform?

-
-
-

### CODE REVIEW

File: packages/fhir-datastore-hapi-fhir/importer/facility-seeder.js [hunk 1]
```diff
@@ -37,36 +38,101 @@ async function saveBundle(bundle) {
...
+
+async function processBundle() {
   try {
-    await saveBundle(secondLocationData); // Call saveBundle function with the parsed bundle
+    const locationData = await fetchDataFromUrl("https://mfldit.gov.org.bw/api/v1/mfl/fhir/bundle/location");
+    const locationResources = await  processBundleEntriesAndGenerateTransactionBundle(locationData);
+
+    await saveBundle(locationResources); // Call saveBundle function with the parsed bundle
+
+    const organizationData = await fetchDataFromUrl("https://mfldit.gov.org.bw/api/v1/mfl/fhir/bundle/organization");
```
Comment (marrouchi) [inline packages/fhir-datastore-hapi-fhir/importer/facility-seeder.js:131]: Let's add env vars

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: packages/fhir-datastore-hapi-fhir/importer/facility-seeder.js (hunk 1 above)
Comment (MohamedAliBouhaouala) [reply packages/fhir-datastore-hapi-fhir/importer/facility-seeder.js]: Done

File: packages/fhir-datastore-hapi-fhir/importer/facility-seeder.js [hunk 2]
```diff
@@ -37,36 +41,104 @@ async function saveBundle(bundle) {
...
+  for (const item of data.entry) {
+    // Fetch the resource from the URL
+    let resource = await fetchDataFromUrl(item.fullUrl);
+
+    // Assign identifiers if the resource is a Location or Organization
+    if (!resource && item.resource.resourceType === 'Location') {
+      resource = item.resource;
+      resource.identifier = [{
+        use: "official",
+        system: "http://moh.bw.org/ext/identifier/mfl-code",
```
Comment (ItsMurumba) [inline packages/fhir-datastore-hapi-fhir/importer/facility-seeder.js:89]: Let us make the url a configurable env variable.

File: packages/fhir-datastore-hapi-fhir/importer/facility-seeder.js [hunk 3]
```diff
@@ -37,36 +41,104 @@ async function saveBundle(bundle) {
...
+        value: resource.id
+      }];
+    } else if (!resource && item.resource.resourceType === 'Organization') {
+      resource = item.resource;
+      resource.identifier = [{
+        use: "official",
+        system: "https://www.hl7.org/fhir/organization.html",
```
Comment (ItsMurumba) [inline packages/fhir-datastore-hapi-fhir/importer/facility-seeder.js:96]: Let us make the url a configurable env variable.

**Review by ItsMurumba — CHANGES_REQUESTED** [review]
Looks good. Just some minor fixes.

**Comment by ItsMurumba** [comment]
Please change the destination branch to main since [fix/omang-env-var](https://github.com/BWHIE/hie-botswana/tree/fix/omang-env-var) is already merged into main.

File: packages/fhir-datastore-hapi-fhir/importer/facility-seeder.js (hunk 2 above)
Comment (MohamedAliBouhaouala) [reply packages/fhir-datastore-hapi-fhir/importer/facility-seeder.js:89]: Those are code systems from the MFL.

Comment (ItsMurumba) [reply packages/fhir-datastore-hapi-fhir/importer/facility-seeder.js:89]: So they can not change?

**Review by ItsMurumba — APPROVED** [review]

**Pull request merged**